File: oa_questions/leetcode/longestCommonPrefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs.length == 0) return "";
        String prefix = strs[0];
        for (int i = 0; i < strs.length; i++) {
            while (strs[i].indexOf(prefix) != 0) {
                prefix = prefix.substring(0, prefix.length() - 1);
                if (prefix.isEmpty()) return "";
            }  
        }
        return prefix;
    }
}


class Solution {
    public String longestCommonPrefix(String[] strs) {
        for (int i = 0; i < strs[0].length(); i++) 
        {
            char tmpChar = strs[0].charAt(i); 
            for (int j = 0; j < strs.length; j++) 
            {
                if (strs[j].length() == i || strs[j].charAt(i) != tmpChar) 
                {
                    return strs[0].substring(0, i);
                }
            }
        }
        return strs[0]; 
    }
}

File: oa_questions/leetcode/longestCommonPrefix.js
/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function (strs) {
  if (strs.length === 0) return "";
  let p = strs[0];
  for (let i = 0; i < strs.length; i++) {
    let pre = "";
    for (let s = 0; s < p.length; s++) {
      if (p.charAt(s) === strs[i].charAt(s)) {
        pre += p.charAt(s);
        continue;
      }
      if (s === 0) return "";
      break;
    }
    p = pre;
  }

  return p;
};
